.circleci/config.yml

```diff
@@ -295,14 +320,27 @@ commands:
           when: always
 
   halt-puppeteer-check:
-    description: "Stop running Puppeteer end-to-end test if conditions are met"
+    description: "Determine whether to skip puppeteer test."
+    parameters:
+      branch:
+        type: string
+        default: 'pr' # pr or master
     steps:
-      - run:
-          name: "Check commit message and find changed file names that do not match ignore patterns"
-          command: bash .circleci/skip-e2e-pr-checks.sh
-      - halt-test-check:
-          # Stop running Puppeteer end-to-end test if code in relevant directories is unchanged.
-          dir_names: "ui e2e .circleci"
+      - when:
+          condition:
+            equal: [ "pr", << parameters.branch >> ]
+          steps:
+            - run:
+                name: "Check commit message and find changed file names that do not match ignore patterns on PR branch."
+                command: bash .circleci/skip-e2e-pr-checks.sh
+      - when:
+          condition:
+            equal: [ "master", << parameters.branch >> ]
+          steps:
+            - run:
+                name: "Check for any non-e2e files that were changed in last commit on master branch."
+                command: bash .circleci/skip-e2e-master-checks.sh
```

Comment:
I would prefer this ran on every master merge - or whenever we run the release, since it can fail due to environmental issues or non-obvious interactions. However, if this is being used as a step to mitigate concurrent runs, then I'm fine adding this temporarily. Running Puppeteer on a fixed interval would be another option.